common/client-libs/mixnet-client/src/forwarder.rs

```diff
@@ -40,12 +42,38 @@ impl PacketForwarder {
             PacketForwarder {
                 mixnet_client: Client::new(client_config),
                 packet_receiver,
+                shutdown,
             },
             packet_sender,
         )
     }
 
     pub async fn run(&mut self) {
+        while !self.shutdown.is_shutdown() {
+            tokio::select! {
+                biased;
+                _ = self.shutdown.recv() => {
+                    log::trace!("PacketForwarder: Received shutdown");
+                }
+                Some(mix_packet) = self.packet_receiver.next() => {
```

Comment:
what about making it stop on channel close? In many cases I suspect that tasks might not even need a `TaskClient` if they stop on channel close

Reply:
good idea

Reply:
while that could work, I think having explicit shutdown check here would be better as if the request queue has built up on `packet_receiver` channel, it will take quite a while for the channel to close on the receiver side.